### PR TITLE
[WF-454] Copy content to new docs

### PR DESCRIPTION
**Pull request by blurbyte**
## Proposed changes

Major features implemented:

- Documentation for all **Waffles components**
- Documentation for **Design** and **Content** sections
- Custom 404 Page
- Various helper components like **Imports**, **LogoPreview**, etc.
- Disabled some linter rules for new docs
- Loosened up some props checking of a few components

Known issues (to be solved in future PRs):
- Toast component SSR warning
- Resizable Elements doesn’t work
- Props warning for Modal footer children
- Props warning for Markdown header children

Desktop:
components-docs

## Functional Code

- [x] Builds without errors
- [x] Linter passes CI
- [ ] ~~Unit tests written & pass CI~~
- [ ] ~~Integration tests written & pass CI~~
- [x] Tested on [supported browsers](https://support.datacamp.com/hc/en-us/articles/360001541574-Minimum-System-Requirements)

## Documentation

- [ ] ~~Technical docs written~~
- [ ] ~~Structural changes reflected in Readme~~
- [ ] ~~Migration plan for breaking changes~~

## Meets Product Requirement

- [x] Assumptions are met
- [x] Meets acceptance criteria
- [ ] ~~Approved by Designer, Engineer, & PO~~

### CODE REVIEW

**Review by hermansje — COMMENTED** [review]
Looks good!
I looked most closely at the things that are not visible, we can fix other things as we spot them while using the documentation.

File: packages/docs/next-doc-site/components/imports.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  module: Module;
+  path: string;
+};
+
+function Imports({ module, path }: ExportsProps): JSX.Element {
+  const defaultExportName = defaultExportNameFromPath(path);
+
+  return (
+    <Fragment>
+      <Markdown.p>
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/docs/next-doc-site/components/imports.tsx:28]: It is imported that way to be consistent with components displayed by markdown itself (components defined in `markdown-elements.tsx`), so if they will be adjusted changes will be applied automatically.

File: packages/docs/next-doc-site/public/logo/logomark.svg [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/docs/next-doc-site/public/logo/logomark.svg:1]: Good catch. Those SVG are pretty mess I will fix them in next PR.

File: packages/docs/next-doc-site/components/markdown-elements.tsx [hunk 3]
```diff
@@ -8,17 +8,22 @@ import {
...
 export type TextProps = {
   children: string | React.ReactNode;
 };
 
+// Add ID to heading, so it could be bookmarked
 function H2({ children }: TextProps): JSX.Element {
   return (
     <Heading as="h2" multiLine size={600}>
-      {children}
+      <span id={slug(textFromChildren(children))}>{children}</span>
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/docs/next-doc-site/components/markdown-elements.tsx:26]: Yeah, in near future I am planning to make such components more flexible. Today or tomorrow will release Switch component with such approach.

File: packages/react-components/text/src/components/List/index.tsx [hunk 4]
```diff
@@ -58,13 +56,6 @@ const List = (props: ListProps): ReactElement => {
   );
 };
 
-List.propTypes = {
-  children: PropTypes.oneOfType([
-    childrenOfType(ListItem),
-    PropTypes.arrayOf(childrenOfType(ListItem)),
-  ]),
-};
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/react-components/text/src/components/List/index.tsx:66]: Already done.

File: packages/docs/next-doc-site/examples/form-elements/multiple-checkboxes.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      <CheckboxList
+        label="What features should Waffles have?"
+        name="feature-wishlist"
+        onChange={setFeatureWishlist}
+        value={featureWishlist}
+      >
+        <Checkbox value="basic">Basic features</Checkbox>
+        <Checkbox value="wrap">
+          Wrapping text when the option takes quite a few words to clearly
+          explain
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/docs/next-doc-site/examples/form-elements/multiple-checkboxes.tsx:18]: It looks rather bad. For sure will fix it in future.

File: packages/docs/next-doc-site/helpers/component-metadata-by-name.ts [hunk 6]
```diff
@@ -5,6 +5,7 @@ function componentMetadataByName(
   metadata: Metadata,
   name: string,
 ): SingleComponentMetadata | undefined {
+  console.log(metadata)
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/docs/next-doc-site/helpers/component-metadata-by-name.ts]: Already removed.

**Pull request merged**